File: edge-engine/src/main/scala/com/cloudentity/edge/accesslog/AccessLogPersister.java
package com.cloudentity.edge.accesslog;

import com.cloudentity.tools.vertx.bus.VertxEndpoint;
import com.cloudentity.tools.vertx.tracing.TracingContext;

public interface AccessLogPersister {
  @VertxEndpoint
  void persist(TracingContext ctx, AccessLog log);
}

File: edge-engine/src/main/scala/com/cloudentity/edge/accesslog/AccessLogPersister.scala
package com.cloudentity.edge.accesslog

import com.cloudentity.tools.vertx.bus.VertxEndpoint
import com.cloudentity.tools.vertx.tracing.TracingContext

trait AccessLogPersister {
  @VertxEndpoint
  def persist(ctx: TracingContext, log: AccessLog): Unit
}
